src/spyglass/lfp/v1/analytic_signal.py

```diff
@@ -0,0 +1,129 @@
+
+        # Next, select data using defined electrode id(s) and defined valid interval.
+        electrode_list = (LFPElectrodeGroup.LFPElectrode() & key).fetch("electrode_id")
+        electrode_index = np.isin(filtered_band.electrodes.data[:], electrode_list)
+        start = (IntervalList() & key).fetch1("valid_times")[0][0]
+        end = (IntervalList() & key).fetch1("valid_times")[0][-1]
+        timestamps_selected = filtered_band.timestamps[
+            np.logical_and(
+                filtered_band.timestamps >= start, filtered_band.timestamps <= end
+            )
+        ]
+        filtered_band_selected = filtered_band.data[
+            np.logical_and(
+                filtered_band.timestamps >= start, filtered_band.timestamps <= end
+            )
+        ][:, electrode_index]
+
+        # Get analytic signal with the specified method (we're only using Hilbert transform for now)
+        filtered_analytic = np.empty(
+            shape=(filtered_band_selected.shape[0], 0), dtype=np.complex_
+        )
+        for lfp_electrode_index in range(filtered_band_selected.shape[1]):
+            filtered_lfp_one_channel = np.array(filtered_band_selected)[
+                :, lfp_electrode_index
+            ]
+            analytic_signal_one_channel = hilbert(filtered_lfp_one_channel, axis=0)
+            filtered_analytic = np.column_stack(
+                (filtered_analytic, analytic_signal_one_channel)
+            )
+
+        # Combine times stamps and analytic signal results into a dataframe
+        timestamps_selected_df = pd.DataFrame({"time stamps": timestamps_selected})
+        analytic_signal_df = pd.DataFrame(
+            filtered_analytic,
+            columns=[f"electrode {e}" for e in electrode_list],
+        )
+        analytic_signal_results = pd.concat(
+            (timestamps_selected_df, analytic_signal_df), axis=1
+        )
+
+        # Create an analysis nwb file to save the results (analytic signal and corresponding time stamps)
+        analytic_signal_file_name = AnalysisNwbfile().create(key["nwb_file_name"])
+        key["analysis_file_name"] = analytic_signal_file_name
+
+        # get object id while generating the nwb file
+        analytic_signal_object_id = AnalysisNwbfile().add_nwb_object(
+            analysis_file_name=key["analysis_file_name"],
+            nwb_object=analytic_signal_results,
+        )
+        AnalysisNwbfile().add(key["nwb_file_name"], analytic_signal_file_name)
+        key["analytic_signal_object_id"] = analytic_signal_object_id
+        self.insert1(key)
+        print(
+            f"\nPopulated analytic signal table for nwb_file_name={key['nwb_file_name']} between {start}ms and {end}ms"
+        )
+
+    def fetch_nwb(self, *attrs, **kwargs):
+        return fetch_nwb(
+            self, (AnalysisNwbfile, "analysis_file_abs_path"), *attrs, **kwargs
+        )
+
+    def fetch1_dataframe(self):
+        return self.fetch_nwb()[0]["analytic_signal"]
```

Comment:
```suggestion
        return self.fetch_nwb()[0]["analytic_signal"].set_index("time stamps")
```
Or whatever your time column is called. (may need to add an inplace=True arg to the set_index call)

Reply:
I think you need to wrap this in a dataframe as well?

Reply:
I think if it's added to scratch space then if you fetch it will return the df. I think this [example](https://github.com/LorenFrankLab/spyglass/blob/011dcaaca1f1dc20f1cc75c0edd8f2a85ad48ce4/src/spyglass/position/v1/position_dlc_pose_estimation.py#L145) works.

Reply:
Ah right. I forgot that this could return a dataframe from scratch.